pkg/i2gw/providers/apisix/http_to_https_test.go

```diff
@@ -0,0 +1,287 @@
+							},
+						},
+					},
+				},
+			},
+			expectedHTTPRoute: &gatewayv1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress-example-com",
+					Namespace: "default",
+				},
+				Spec: gatewayv1.HTTPRouteSpec{
+					Hostnames: []gatewayv1.Hostname{"example.com"},
+					Rules: []gatewayv1.HTTPRouteRule{
+						{
+							BackendRefs: []gatewayv1.HTTPBackendRef{
+								{BackendRef: gatewayv1.BackendRef{BackendObjectReference: gatewayv1.BackendObjectReference{Name: "example", Port: ptr.To(gatewayv1.PortNumber(3000))}}},
+							},
+						},
+					},
+				},
+			},
+			expectedError: field.ErrorList{},
+		},
+		{
+			name: "annotation present but invalid value",
+			ingress: networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"k8s.apisix.apache.org/http-to-https": "invalid",
+					},
+				},
+				Spec: networkingv1.IngressSpec{
+					Rules: []networkingv1.IngressRule{
+						{
+							Host: "example.com",
+						},
+					},
+				},
+			},
+			initialHTTPRoute: &gatewayv1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress-example-com",
+					Namespace: "default",
+				},
+				Spec: gatewayv1.HTTPRouteSpec{
+					Hostnames: []gatewayv1.Hostname{"example.com"},
+					Rules: []gatewayv1.HTTPRouteRule{
+						{
+							BackendRefs: []gatewayv1.HTTPBackendRef{
+								{BackendRef: gatewayv1.BackendRef{BackendObjectReference: gatewayv1.BackendObjectReference{Name: "example", Port: ptr.To(gatewayv1.PortNumber(3000))}}},
+							},
+						},
+					},
+				},
+			},
+			expectedHTTPRoute: &gatewayv1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress-example-com",
+					Namespace: "default",
+				},
+				Spec: gatewayv1.HTTPRouteSpec{
+					Hostnames: []gatewayv1.Hostname{"example.com"},
+					Rules: []gatewayv1.HTTPRouteRule{
+						{
+							BackendRefs: []gatewayv1.HTTPBackendRef{
+								{BackendRef: gatewayv1.BackendRef{BackendObjectReference: gatewayv1.BackendObjectReference{Name: "example", Port: ptr.To(gatewayv1.PortNumber(3000))}}},
+							},
+						},
+					},
+				},
+			},
+			expectedError: field.ErrorList{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ingressResources := i2gw.InputResources{
+				Ingresses: []networkingv1.Ingress{tc.ingress},
+			}
+			gatewayResources := &i2gw.GatewayResources{
+				HTTPRoutes: map[types.NamespacedName]gatewayv1.HTTPRoute{
+					{Name: tc.expectedHTTPRoute.Name, Namespace: tc.expectedHTTPRoute.Namespace}: *tc.initialHTTPRoute,
+				},
+			}
+
+			errs := httpToHttpsFeature(ingressResources, gatewayResources)
+
+			if len(errs) != len(tc.expectedError) {
+				t.Fatalf("expected %d errors, got %d", len(tc.expectedError), len(errs))
```

Comment:
why Fatal? I think `t.Errorf` is more appropriate. 

ditto for all `t.Fatalf` mentions in this file

Reply:
I referred to the test cases of other providers. Changed now.